src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,513 @@
+  - `DPoPValidationMode` is a non-nullable column that controls the DPoP validation mechanism. Existing clients that are not using DPoP can set its value to `0`.
+  - `DPoPClockSkew` is a non-nullable timespan that controls how much clock skew is allowed for a particular DPoP client. Existing clients that are not using DPoP can set its value to a timespan of length ``0.
+
+- The `ServerSideSession` entity now uses a 64-bit long as its primary key.
+- Two new properties have been added to the `Client` model:
+  - `Client.RequirePushedAuthorization` is a new boolean property that controls if this client requires [pushed authorization requests (PAR)](../../tokens/par). It is safe to initialize this column to `false` for existing clients, which will mean that the global PAR configuration will be used.
+  - `Client.PushedAuthorizationLifetime` is a new nullable integer property that controls the lifetime of pushed
+    authorization requests (in seconds) for a client. It is safe to initialize this column to `null` for existing clients, which means the global value is used.
+- A new `PushedAuthorizationRequest` table has been added to store pushed authorization requests.
+
+In this section, we'll look at updating the database schema based on the stores provided in the `Duende.IdentityServer.EntityFramework` package.
+
+You'll need to create two database migrations that update the database schema: one that targets the `PersistedGrantDbContext` (for operational data), and one that targets the `ConfigurationDbContext` (for configuration data).
+Note that you may want to change the database migration paths in the examples below to match your project structure.
+
+<Steps>
+
+1. Create the migrations for the operational and configuration database context:
+
+   ```bash title="Terminal"
+   dotnet ef migrations add UpdateToDuende_v7_0 -c PersistedGrantDbContext -o Data/Migrations/IdentityServer/PersistedGrantDb
+   dotnet ef migrations add UpdateToDuende_v7_0 -c ConfigurationDbContext -o Data/Migrations/IdentityServer/ConfigurationDb
+   ```
+
+   :::note
+   You may see a warning _"An operation was scaffolded that may result in the loss of data. Please review the migration for accuracy."_.
+   The column length for redirect URIs (for both login and logout) was reduced from 2000 to 400 to overcome database index size limits.
+   Unless you are using redirect URIs greater than 400 characters, this should not affect you.
+   :::
+
+2. Apply the migrations to your database:
+
+   ```bash title="Terminal"
+   dotnet ef database update -c PersistedGrantDbContext
+   dotnet ef database update -c ConfigurationDbContext
+   ```
+
+</Steps>
+
+### Step 6: Migrate Signing Keys <Badge text="Optional" variant="note" size="small" />
+
+If your IdentityServer4 implementation is using a signing key, consider using [automatic key management](../../fundamentals/key-management).
```

Comment:
I think we should mention that it's a business license feature? I've had too many customers realizing too late that they used a business feature and only bought a starter license.